internal/config.go

```diff
@@ -59,3 +59,77 @@ func ReadConnectionsFile(path string) ConnectionsConfigFile {
 	}
 	return config
 }
+
+// ValidateConfigFile checks that the config file does not contain any unknown properties
+// and then, if the config is valid, reads it.
+func ValidateConfigFile(explicitConfigFile string) {
+	configFile := "" // Just for logging
+	if explicitConfigFile != "" {
+		explicitConfigFile = strings.TrimSpace(explicitConfigFile)
+		validateProps(explicitConfigFile)
+		viper.SetConfigFile(explicitConfigFile)
+		if err := viper.ReadInConfig(); err == nil {
+			configFile = explicitConfigFile
+			LogVerbose("Using config file: " + explicitConfigFile)
+		} else {
+			FatalError(err)
+		}
+	} else {
+		configFile = findConfigFile("corectl") // name of config file (without extension)
+		if configFile != "" {
+			validateProps(configFile)
+			viper.SetConfigFile(configFile)
+			if err := viper.ReadInConfig(); err != nil {
+				FatalError("Failed to read config file " + configFile)
+			}
+		}
+	}
+	QliVerbose = viper.GetBool("verbose")
+	LogTraffic = viper.GetBool("traffic")
+	if configFile != "" {
+		LogVerbose("Using config file: " + configFile)
+	} else {
+		LogVerbose("No config file")
+	}
+
+}
+
+// validateProps reads a config file by 
+func validateProps(configPath string) {
+	source, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		FatalError("Could not find config file:", configPath)
+	}
+	validProps := map[string]struct{}{ // This "set" contains the valid property names
```

Comment:
Drive-by review, hope you don't mind :)

Can we get this list from somewhere? I'm afraid this might get out of sync easily when we add new properties?

LGTM otherwise, good work!

Reply:
Also, a fun idea for another time perhaps. It's common to find recommendations in CLI for typos, e.g. `trafic` and `traffic`: `Did you mean 'traffic'`? You can easily calculate that using [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) :)

Reply:
Good point. I was thinking of solving this solution using `viper.AllSettings()`, but, this method apparently doesn't return all possible properties.

Gabriel suggested to solve this problem following the merge of his noun-verb branch where he has also collected the initialization of all flags and props to one file. From this file we can easily create the set of valid properties.

So maybe we could use this as a temporary solution and I can fix it properly following the merge of Gabriel's branch.

Reply:
And yes, that would be a nice feature, but probably beyond the scope of this PR ^^
I will add it as an enhancement issue!